streams/src/test/java/org/apache/kafka/streams/errors/AlwaysContinueProductionExceptionHandler.java

```diff
@@ -1,44 +0,0 @@
-/*
- * Licensed to the Apache Software Foundation (ASF) under one or more
- * contributor license agreements. See the NOTICE file distributed with
- * this work for additional information regarding copyright ownership.
- * The ASF licenses this file to You under the Apache License, Version 2.0
- * (the "License"); you may not use this file except in compliance with
- * the License. You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.apache.kafka.streams.errors;
-
-import org.apache.kafka.clients.producer.ProducerRecord;
-
-import java.util.Map;
-
-/**
- * Production exception handler that always instructs streams to continue when an exception
- * happens while attempting to produce result records.
- */
-public class AlwaysContinueProductionExceptionHandler implements ProductionExceptionHandler {
```

Comment:
Why does this PR remove this class?

Reply:
@mjsax 

We have created a parameterized `ProductionExceptionHandler` (`ProductionExceptionHandlerMock`) in the `RecordCollectorTest` class that additionally does some assertions, including the origin of the exception. 

Thus `AlwaysContinueProductionExceptionHandler` was redundant with `ProductionExceptionHandlerMock`, it has been replaced in this PR while updating and adding some additional unit tests

Reply:
But `AlwaysContinueProductionExceptionHandler` is a public built-in handler users can configure... -- it's public API, and thus has nothing to do with `ProductionExceptionHandlerMock` and our internal testing.

It's a build-in alternative to the default handler.

If we really want to no have it any longer, we would need to deprecate -- but we cannot just remove public API. But I don't see why we would want to deprecate and remove it?

Reply:
@mjsax Unless I'm wrong, `AlwaysContinueProductionExceptionHandler` belongs to `src/test/java` and is only used for testing purpose in the `RecordCollectorTest`. 

Given that it is only use for testing, should it be consider public? If yes, we can obviously restore the class

Reply:
Ah. You are right. My bad. I guess I mixed it up with the two built-in handler for `DeserializationExceptionHandler`, but we only have the "alway fail" build-in handler for the production case...